uncertainty_baselines/models/radial_utils.py

```diff
@@ -0,0 +1,447 @@
+    """Constructs the initializer."""
+    super(TrainableRadial, self).__init__(**kwargs)
+    self.mean_initializer = get(mean_initializer)
+    self.stddev_initializer = get(stddev_initializer)
+    self.mean_regularizer = regularizers.get(mean_regularizer)
+    self.stddev_regularizer = regularizers.get(stddev_regularizer)
+    self.mean_constraint = constraints.get(mean_constraint)
+    self.stddev_constraint = constraints.get(stddev_constraint)
+    self.seed = seed
+
+  def build(self, shape, dtype=None):
+    if dtype is None:
+      dtype = self.dtype
+
+    self.mean = self.add_weight(
+      'mean',
+      shape=shape,
+      initializer=self.mean_initializer,
+      regularizer=self.mean_regularizer,
+      constraint=None,
+      dtype=dtype,
+      trainable=True)
+    self.stddev = self.add_weight(
+      'stddev',
+      shape=shape,
+      initializer=self.stddev_initializer,
+      regularizer=self.stddev_regularizer,
+      constraint=None,
+      dtype=dtype,
+      trainable=True)
+    self.built = True
+
+  def __call__(self, shape, dtype=None):
+    if not self.built:
+      self.build(shape, dtype)
+    mean = self.mean
+    if self.mean_constraint:
+      mean = self.mean_constraint(mean)
+    stddev = self.stddev
+    if self.stddev_constraint:
+      stddev = self.stddev_constraint(stddev)
+    return generated_random_variables.make_random_variable(Radial)(
+      loc=mean,
+      presoftplus_scale=tf.math.log(tf.math.expm1(stddev)))
+
+  def get_config(self):
+    return {
+      'mean_initializer':
+        serialize(self.mean_initializer),
+      'stddev_initializer':
+        serialize(self.stddev_initializer),
+      'mean_regularizer':
+        regularizers.serialize(self.mean_regularizer),
+      'stddev_regularizer':
+        regularizers.serialize(self.stddev_regularizer),
+      'mean_constraint':
+        constraints.serialize(self.mean_constraint),
+      'stddev_constraint':
+        constraints.serialize(self.stddev_constraint),
+      'seed': self.seed,
+    }
+
+
+class TrainableHeRadial(TrainableRadial):
+  """Trainable radial initialized per He et al. 2015, given a ReLU nonlinearity.
+
+  The distribution is initialized to a Radial scaled by `sqrt(2 / fan_in)`,
+  where `fan_in` is the number of input units. A ReLU nonlinearity is assumed
+  for this initialization scheme.
+
+  References:
+    He K, Zhang X, Ren S, Sun J. Delving deep into rectifiers: Surpassing
+    human-level performance on imagenet classification. In Proceedings of the
+    IEEE international conference on computer vision 2015 (pp. 1026-1034).
+    https://arxiv.org/abs/1502.01852
+  """
+
+  def __init__(self, seed=None, **kwargs):
+    super(TrainableHeRadial, self).__init__(
+      mean_initializer=tf.keras.initializers.he_normal(seed),
+      seed=seed,
+      **kwargs)
+
+  def get_config(self):
+    return {
+      'seed': self.seed,
+    }
+
+
+class RadialKLDivergenceWithTiedMean(tf.keras.regularizers.Regularizer):
```

Comment:
Agreed - I added this for comparison to the VI implementation, which has support for a MVG prior with tied mean. 
For the time being (while we don't have a Radial-Radial KL) I'll leave these Radial regularizers (means tied and otherwise) out.

Reply:
Makes sense.